### PR TITLE
added a function to get set of users who have sent a message

### PR DESCRIPTION
**Pull request by sirisjo**

### CODE REVIEW

**Comment by sirisjo** [comment]
not sure why it thinks I deleted everything, I just added my 1 method to the bottom.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -1,99 +1,111 @@
...
+   * Gets messages posted by a specific user.
+   *
+   * @return a list of messages posted by the user, or empty list if user has never posted a
+   * message. List is sorted by time descending.
+   */
+  public List<Message> getMessages(String user) {
+    Query query =
+            new Query("Message")
+                    .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
+                    .addSort("timestamp", SortDirection.DESCENDING);
```
Comment (matthew-chinn) [inline src/main/java/com/google/codeu/data/Datastore.java]: Can we make this indentation either 4 spaces or 2? I think Google's Java style guide says 4 spaces if we are continuing the same line, but 2 spaces if it is a new code block (such as after {}). However, we will try using a formatter later to do this for us, but for now lets make it somewhat consistent with the rest!

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -1,99 +1,111 @@
...
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Message> messages = getMessageInformation(results);
+
+    return messages;
+  }
+
+  public List<Message> getAllMessages() {
+    Query query = new Query("Message")
+            .addSort("timestamp", SortDirection.DESCENDING);
```
Comment (matthew-chinn) [inline src/main/java/com/google/codeu/data/Datastore.java]: Same here :)

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 3]
```diff
@@ -1,99 +1,111 @@
...
+        System.err.println("Error reading message.");
+        System.err.println(entity.toString());
+        e.printStackTrace();
+      }
+    }
+    return messages;
+  }
+
+  public Set<String> getUsers() {
+	  Set<String> users = new HashSet<>();
```
Comment (matthew-chinn) [inline src/main/java/com/google/codeu/data/Datastore.java]: Please line up this indentation!

**Review by matthew-chinn — CHANGES_REQUESTED** [review]

**Comment by sirisjo** [comment]
I think I committed the same thing twice, oops...

**Review by matthew-chinn — APPROVED** [review]
Thanks! lgtm

**Pull request merged**